src/Stratis.Bitcoin.Features.Consensus/ConsensusController.cs

```diff
@@ -45,6 +49,36 @@ public uint256 GetBestBlockHashRPC()
             return this.ChainState.ConsensusTip?.HashBlock;
         }
 
+        /// <summary>
+        /// Get the threshold states of softforks currently being deployed.
+        /// Allowable states are: Defined, Started, LockedIn, Failed, Active.
+        /// </summary>
+        /// <returns>Json formatted type with Deployment Index <see cref="int"/>, 
+        /// State Value <see cref="ThresholdState"/>, human readable Threshold State <see cref="string"/>
+        /// Returns <see cref="IActionResult"/> formatted error if fails.
+        /// </returns>.
+        [Route("api/[controller]/deploymentflags")]
+        [HttpGet]
+        public IActionResult DeploymentFlags()
+        {
+            try
+            {
+               var rule = this.ConsensusManager.ConsensusRules.GetRule<SetActivationDeploymentsFullValidationRule>();
```

Comment:
Why getting a rule if we need rule engine?

Reply:
Updated to rule engine.